alibi_detect/cd/pytorch/mmd.py

```diff
@@ -89,7 +89,7 @@ def __init__(
         # initialize kernel
         sigma = torch.from_numpy(sigma).to(self.device) if isinstance(sigma,  # type: ignore[assignment]
                                                                       np.ndarray) else None
-        self.kernel = kernel(sigma) if kernel == GaussianRBF else kernel
+        self.kernel = kernel(sigma).to(self.device) if kernel == GaussianRBF else kernel
```

Comment:
Seems like a bug, probably worth opening a separate PR to get this in `master` quickly and not clutter this PR?